### ISSUE TITLE
prefer-writable-derived is misleading with sveltekit

### ISSUE THREAD

**Issue by RSWilli** [sveltejs/eslint-plugin-svelte#1446]
The [eslint-plugin-svelte docs](https://sveltejs.github.io/eslint-plugin-svelte/rules/prefer-writable-derived/) mention the following:

```svelte
<script>
  /* eslint svelte/prefer-writable-derived: "error" */
  const { initialValue } = $props();

  // ✓ GOOD
  let value1 = $derived(initialValue);

  // ✗ BAD
  let value2 = $state(initialValue);
  $effect(() => {
    value2 = initialValue;
  });
</script>
```

Which is fine for simple values like numbers and strings, because they are copied.

It can lead to bugs though once you start mutating the value. Because [$derived does not convert to deeply reactive proxies](https://svelte.dev/docs/svelte/$derived#Deriveds-and-reactivity) (unlike $state), the derived value may not trigger UI updates as expected.

This happens especially when the source value isn't a $state, e.g. when coming from sveltekit's `PageData`

```svelte
<script>
  import type { PageData } from "./$types"

  const { data } = $props<{ data: PageData }>();

  let { initialValue } = $derived(data)

  // value is not deeply reactive, because initialValue isn't. 
  let value = $derived(initialValue);
</script>
```

An example is the basic loading tutorial: https://svelte.dev/tutorial/kit/page-data if the loaded data is only the initial state for an interactive application, then $derive-ing will have implications that are hard to debug.

------

This may also be a sveltekit issue. I spent an annoying amount of time debugging why my UI didn't update, only to find that the solution is to not follow the `prefer-writable-derived` linter rule.

**Comment by sacrosanctic** [comment]
related https://github.com/sveltejs/svelte/pull/17308
